src/Sarif.Viewer.VisualStudio.ResultSources.DeveloperCanvas.Core/Services/DevCanvasResultSourceService.cs

```diff
@@ -0,0 +1,384 @@
+                    }
+
+                    // Periodically log a message to indicate how many files are still queued for processing.
+                    queuedCount = filePathQueue.Count;
+                    if (queuedCount > 0 && (queuedCount % 100 == 0))
+                    {
+                        resultMessage.Append($"\n{queuedCount} files remaining.");
+                    }
+                }
+
+                // Display the result message (e.g. "Cached X insights for file Y")
+                Trace.WriteLine(resultMessage.ToString());
+
+                RaiseServiceEvent(new ResultsUpdatedEventArgs()
+                {
+                    SarifLog = masterLog,
+                    LogFileName = "",
+                    UseDotSarifDirectory = false
+                });
+            }
+            catch (Exception)
+            {
+            }
+        }
+
+        /// <inheritdoc/>
+        public async Task<Result<bool, ErrorType>> RequestAnalysisScanResultsAsync(object data = null)
+        {
+            // called from result source host's RequestAnalysisResultsAsync
+            // called on a repo-basis, not a file/folder basis and we do not want to scan an entire repo at once for performance reasons and so we will leave it as-is.
+            return Result.Success<bool, ErrorType>(true);
+        }
+
+        /// <summary>
+        /// Gets the repo rooted version of the absolute file path.
+        /// </summary>
+        /// <param name="gitRepoRoot">Root of the code base.</param>
+        /// <param name="absoluteFilePath">The file path we want a relative file path of</param>
+        /// <returns>Repo rooted file path starting with the name of the directory.</returns>
+        /// <exception cref="Exception">Thrown when the <paramref name="gitRepoRoot"/> is not part of <paramref name="absoluteFilePath"/>'s path.</exception>
+        private string GetRepoRootedFilePath(string gitRepoRoot, string absoluteFilePath)
+        {
+            gitRepoRoot = gitRepoRoot.Replace("/", "\\");
+            string correctedAbsPath = absoluteFilePath.Replace("/", "\\");
+            if (correctedAbsPath.StartsWith(gitRepoRoot))
+            {
+                string rootedFilePath = correctedAbsPath.Substring(gitRepoRoot.Length);
+                if (rootedFilePath.StartsWith("\\"))
+                {
+                    rootedFilePath = rootedFilePath.Substring(1);
+                }
+                return rootedFilePath;
+            }
+            else
+            {
+                throw new Exception($"Repo root: {gitRepoRoot} is not a part of {correctedAbsPath}.");
+            }
+        }
+
+        private const string XamlMessageKey = "Xaml";
+
+        /// <summary>
+        /// Trims the log to make a lighter log that can be processed easier.
+        /// </summary>
+        /// <param name="log"></param>
+        /// <returns></returns>
+        private SarifLog TrimLog(SarifLog log)
```

Comment:
If this is to remove xaml, I'd just name it exactly that to make it more obvious rather than "TrimLog" - to me that sounds like removing whitespace.